cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -170,13 +172,21 @@ func (m *AwsManager) GetAsgForInstance(instance *AwsRef) (*Asg, error) {
 	if config, found := m.asgCache[*instance]; found {
 		return config, nil
 	}
+	if _, found := m.instancesNotInManagedAsg[*instance]; found {
+		// The instance is already known to not belong to any configured ASG
+		// Skip regenerateCache so that we won't unnecessarily call DescribeAutoScalingGroups
+		// See https://github.com/kubernetes/contrib/issues/2541
```

Comment:
Let's link to the issue in the new repo instead: https://github.com/kubernetes/autoscaler/issues/45